### DISCUSSION TITLE
Allow modding of the minimum time to aim before AI can attack with a weapon if they could not see the target initially

### DISCUSSION THREAD

**Discussion by 4-S-H** [FakeFishGames/Barotrauma#14828] [Code questions and modding support]
https://github.com/FakeFishGames/Barotrauma/blob/0e8fb6569d2810e2f8ad5fb17b4bba546cc5739a/Barotrauma/BarotraumaShared/SharedSource/Characters/AI/EnemyAIController.cs#L2432-L2442
On line 2434, it would be good if modders can change monsters' "reaction time".

**Comment by SomeRandomNoobKekeke** [comment]
I don't think thats "reaction time"
I don't see the whole picture here but you can set [aim speed for humans](https://github.com/FakeFishGames/Barotrauma/blob/master/Barotrauma/BarotraumaShared/SharedSource/Characters/AI/Objectives/AIObjectiveCombat.cs#L139-L140) and it's not reaction time, it's time between grabbing a gun and pulling the trigger
I don't think it's applicable to monsters

**Reply by 4-S-H** [reply]
it is checked only when the ai already has an item (ie gun) equipped though
